app/helpers/validation/profile.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+import logging
+
+import geojson
+from flask import abort
+from shapely.geometry import shape
+
+from app import make_error_msg
+from app.helpers.profile_helpers import PROFILE_MAX_AMOUNT_POINTS, PROFILE_DEFAULT_AMOUNT_POINTS
+from app.helpers.validation import srs_guesser
+
+logger = logging.getLogger(__name__)
+
+
+def read_linestring(request_object):
+    # param geom, list of coordinates defining the line on which we want a profile
+    linestring = None
+    if 'geom' in request_object.args:
+        linestring = request_object.args.get('geom')
+    elif request_object.is_json and len(request_object.get_data(as_text=True)) > 0:
+        linestring = request_object.get_json()
+    if not linestring:
+        abort(400, "No 'geom' given, cannot create a profile without coordinates")
+    try:
+        geom = geojson.loads(linestring, object_hook=geojson.GeoJSON.to_instance)
+    except ValueError:
+        raise abort(400, "Error loading geometry in JSON string")
+    try:
+        geom_to_shape = shape(geom)
+    except Exception as e:
+        logger.exception(e)
+        raise abort(400, "Error converting JSON to Shape")
+    try:
+        geom_to_shape.is_valid
+    except Exception:
```

Comment:
Consider
```python
if not geom_to_shape.is_valid:
    abort(400, "Invalid Linestring syntax")
```

Reply:
I tried this way, but it changed how the code reacted (many unit tests failing), so in the end I kept it as it was